react/SelectBox/SelectBox.jsx

```diff
@@ -48,6 +50,44 @@ const customStyles = props => ({
   })
 })
 
+/**
+ * Determines maxHeight property for menuList component. This value is computed
+ * with a container element and the current SelectBox element.
+ * The container element defines an element which the SelectBox should not
+ * overflow.
+ * @param  {object} container React reference to an element containing
+ * the SelectBox
+ * @param  {object} element    React reference to the ReactSelect element
+ * @return {object}            A style object, with `menuList` property set.
+ */
+const computedMenuListHeightStyles = (container, selectElement) => {
+  if (!(container && selectElement)) return {}
+
+  const isPreactRef = !container.current
+  const actualContainer = isPreactRef ? container : container.current
```

Comment:
Nit: I personally prefer to shadow `container` in those cases. This makes it clear that we are not going to use `container` afterwards. And it makes it simpler to reason about this function since the number of variable is reduced.

```
container = container.current || container
```

Nit: Since this function's responsibility is about computing heights, I would not put it in charge of knowing the difference between preact and react.

Reply:
It is also bothering me. I will externalize this part, thanks.